models/issue.go

```diff
@@ -1153,6 +1155,14 @@ func (opts *IssuesOptions) setupSession(sess *xorm.Session) {
 			}
 		}
 	}
+
+	if len(opts.IncludedLabelNames) > 0 {
+		sess.In("issue.id", BuildLabelNamesIssueIDsCondition(opts.IncludedLabelNames))
```

Comment:
@guillep2k can we do better here? I'm suspicious that this `In` could be quite expensive.

Reply:
`WHERE EXISTS (SELECT ...)` is the equivalent idiom, but depending on the engine it could be better optimized.

Reply:
It all depends on how sophisticated the engine is. Some engines will rearrange the whole query and be very clever on optimizing the access... others will just take your statement more or less literally, and therefore your syntax might influence the results drastically. And versions are important too. I read from time to time "XX SQL now takes this construct and treats it like xxx for better performance".

Formally, `WHERE EXISTS (SELECT ...)` is equivalent to `IN (SELECT ....)`; but if the engine is not too smart, `EXISTS` is a kind of join that is more optimizable than a plain LOOP JOIN (e.g. using a HASH JOIN).

Reply:
But performing the `IN()` in a subquery is definitely more performant that `SELECT ids`, and then `SELECT ... WHERE ... IN(id list).

Reply:
Like @guillep2k said, it depends on the database engine. The sub query maybe cached or not. It may be faster than before or not. But it will give the right result however many there are labels. But for `In` with `id`, when there are over 500 variables, database will return error.